wen-restart/src/wen_restart.rs

```diff
@@ -531,91 +578,275 @@ mod tests {
+                    last_voted_fork_slots: init_result.last_voted_fork_slots.clone(),
+                    last_vote_bankhash: last_vote_bankhash.to_string(),
+                    shred_version: SHRED_VERSION as u32,
+                    wallclock: start_time,
+                }),
+                last_voted_fork_slots_aggregate: Some(LastVotedForkSlotsAggregateRecord {
+                    received: HashMap::new()
+                }),
             }
+        )
+        .is_ok());
+        let mut rng = rand::thread_rng();
+        let mut expected_messages = HashMap::new();
+        let expected_slots_to_repair: Vec<Slot> =
+            (last_vote_slot + 1..last_vote_slot + 3).collect();
+        // Skip the first 2 validators, because 0 is myself, we need 8 so it hits 80%.
+        for keypairs in init_result.validator_voting_keypairs.iter().skip(2) {
+            let wen_restart_proto_path_clone = init_result.wen_restart_proto_path.clone();
+            let cluster_info_clone = init_result.cluster_info.clone();
+            let blockstore_clone = init_result.blockstore.clone();
+            let bank_forks_clone = init_result.bank_forks.clone();
+            let exit = Arc::new(AtomicBool::new(false));
+            let exit_clone = exit.clone();
+            let wen_restart_thread_handle = Builder::new()
+                .name("solana-wen-restart".to_string())
+                .spawn(move || {
+                    assert!(wait_for_wen_restart(
+                        &wen_restart_proto_path_clone,
+                        VoteTransaction::from(Vote::new(vec![last_vote_slot], last_vote_bankhash)),
+                        blockstore_clone,
+                        cluster_info_clone,
+                        bank_forks_clone,
+                        Some(Arc::new(RwLock::new(Vec::new()))),
+                        80,
+                        exit_clone,
+                    )
+                    .is_err());
+                })
+                .unwrap();
+            let node_pubkey = keypairs.node_keypair.pubkey();
+            let node = LegacyContactInfo::new_rand(&mut rng, Some(node_pubkey));
+            let last_vote_hash = Hash::new_unique();
+            let now = timestamp();
+            push_restart_last_voted_fork_slots(
+                init_result.cluster_info.clone(),
+                &node,
+                &expected_slots_to_repair,
+                &last_vote_hash,
+                &keypairs.node_keypair,
+                now,
+            );
+            expected_messages.insert(
+                node_pubkey.to_string(),
+                LastVotedForkSlotsRecord {
+                    last_voted_fork_slots: expected_slots_to_repair.clone(),
+                    last_vote_bankhash: last_vote_hash.to_string(),
+                    shred_version: SHRED_VERSION as u32,
+                    wallclock: now,
+                },
+            );
+            sleep(Duration::from_millis(200));
```

Comment:
Maybe a comment as well about what you're waiting to happen, i.e. waiting for the wen_restart thread to detect the message you pushed to gossip via `push_restart_last_voted_fork_slots` above

Reply:
Maybe a comment as well about what you're waiting to happen, i.e. waiting for the wen_restart thread to detect the message you pushed to gossip via `push_restart_last_voted_fork_slots` above

Reply:
Changed.